addon/addon.go

```diff
@@ -204,13 +204,25 @@ func NewAddonManager(kubeConfig *rest.Config) (addonmanager.AddonManager, error)
 		klog.Errorf("unable to setup addon manager: %v", err)
 		return nil, err
 	}
+	addonClient, err := addonv1alpha1client.NewForConfig(kubeConfig)
+	if err != nil {
+		klog.Errorf("unable to setup addon client: %v", err)
+		return nil, err
+	}
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		klog.Errorf("unable to create kube client: %v", err)
-		return addonMgr, err
+		return nil, err
 	}
 	agentAddon, err := addonfactory.NewAgentAddonFactory(SearchAddonName, ChartFS, ChartDir).
-		WithGetValuesFuncs(getValue, addonfactory.GetValuesFromAddonAnnotation).
+		WithConfigGVRs(
+			addonfactory.AddOnDeploymentConfigGVR,
+		).
+		WithGetValuesFuncs(getValue,
+			addonfactory.GetAddOnDeloymentConfigValues(
+				addonfactory.NewAddOnDeloymentConfigGetter(addonClient),
+				addonfactory.ToAddOnNodePlacementValues,
+			), addonfactory.GetValuesFromAddonAnnotation).
```

Comment:
would you put the `addonfactory.GetAddOnDeloymentConfigValues` at the end of the get values functions, like
```
WithGetValuesFuncs(
      getValue, 
      addonfactory.GetValuesFromAddonAnnotation,
      addonfactory.GetAddOnDeloymentConfigValues(
	addonfactory.NewAddOnDeloymentConfigGetter(addonClient),
	addonfactory.ToAddOnNodePlacementValues),
    ),
).
```

thus the addondeployment can override other configs, we may deprecate the addon value annotation (`addonfactory.GetValuesFromAddonAnnotation`) in future

Reply:
Sure . Please let us know when you deprecate , we are using it for clients now .